update-vo-config/update-vo-config

```diff
@@ -0,0 +1,507 @@
+    template.write("'account_prefix' ?= '" + account_prefix + "';\n\n")
+
+    # VOMS Servers
+    template.write("'voms_servers' ?= list(\n")
+    if not vo_details['voms_endpoints']:
+        sys.stderr.write("WARNING: VO " + vo_details['name'] +
+                         " has no VOMS endpoint defined\n")
+    elif len(vo_details['voms_endpoints']) == 1:
+        force_voms_admin = True
+    for endpoint in vo_details['voms_endpoints']:
+        template.write("    dict(\n")
+        template.write("        'name', '%s',\n" % endpoint.hostname)
+        template.write("        'host', '%s',\n" % endpoint.hostname)
+        template.write("        'port', %s,\n" % endpoint.vomses_port)
+        template.write("        'adminport', %s,\n" % endpoint.https_port)
+        if not endpoint.is_admin:
+            if not force_voms_admin:
+                template.write("        'type', list('voms-only'),\n")
+            else:
+                sys.stdout.write("WARNING: voms-admin enabled on " +
+                                 "%s as this is the " % endpoint.hostname +
+                                 "only VOMS server (VO card " +
+                                 "inconsistency)\n")
+        template.write("    ),\n")
+    template.write(");\n\n")
+
+    # VOMS Mapping
+    template.write("'voms_mappings' ?= list(\n")
+    legacy_suffix = []
+    for mapping in vo_details['voms_mapping']:
+        suffix = get_legacy_suffix(vo_details['id'], mapping, legacy_suffix)
+        legacy_suffix.append(suffix)
+        suffix2 = get_account_suffix(vo_details['name'], mapping)
+        template.write("    dict(\n")
+        template.write("        'description', '%s',\n" % mapping['desc'])
+        template.write("        'fqan', '%s',\n" % mapping['expr'])
+        template.write("        'suffix', '%s',\n" % suffix)
+        template.write("        'suffix2', '%s',\n" % suffix2)
+        template.write("    ),\n")
+
+    template.write(");\n\n")
+
+    # Base UID
+    base_uid = vo_details['id'] * 1000
+    template.write("'base_uid' ?= %i;\n" % base_uid)
+    template.close()
+
+
+def update_voms_servers(vo_details, basedir, namespaces, debug=False):
+    """write template containing the voms certficate
+    """
+    voms_servers = {}
+    for vo_config in vo_details:
+        if not vo_config['voms_endpoints']:
+            sys.stderr.write("No VOMS server defined in any VO\n")
+        else:
+            for server in vo_config['voms_endpoints']:
+                if server.hostname not in voms_servers:
+                    voms_servers[server.hostname] = server.cert
+    for (hostname, cert) in voms_servers.items():
+        server_cert_ns = namespaces['cert_dir'] + os.path.sep + hostname
+        server_cert_tpl = basedir + os.path.sep + server_cert_ns + ".pan"
+        if cert is None:
+            sys.stdout.write("No certicate defined for %s " % hostname +
+                             "voms server.\n")
```

Comment:
```python
sys.stdout.write("No certicate defined for %s voms server.\n" % hostname)
```
Why not use use `print`?

Reply:
sys.stdout.write has the same syntax both in Python 2 and Python 3, and the transition will be easier if we have to use a logging module.

Reply:
:+1: